src/micro.c

```diff
@@ -75,11 +99,20 @@ micro_add_endpoint(microEndpoint **new_ep, microService *m, const char *prefix,
     if (cfg == NULL)
         return NULL;
 
+    // retain `m` before the endpoint uses it for its on_complete callback.
```

Comment:
The service now does its own refcounting for endpoints to avoid a race on cleanup